### PR TITLE
Fix connection type parsing

### PR DESCRIPTION
**Pull request by povilasjurcys**
We incorrectly parse connection type, which breaks JSON:API style fieldset config. This PR improves this part by fixing type unwrap logic. Seems like there are more similar issues, but this PR handles only fieldset part.

### CODE REVIEW

File: lib/graphql_to_rest/type_parsers/graphql_type_parser.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ class GraphqlTypeParser
       rattr_initialize %i[unparsed_type!]
 
       def open_api_type_name
-        basic_type_name || graphql_name
+        (basic_type_name || graphql_name)
```
Comment (same-dovydas) [inline lib/graphql_to_rest/type_parsers/graphql_type_parser.rb]: unnecessary change
```suggestion
        basic_type_name || graphql_name
```

**Review by same-dovydas — APPROVED** [review]

**Review by maksim-kulagin-s — APPROVED** [review]
